internal/common/quota.go

```diff
@@ -0,0 +1,60 @@
+package common
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"time"
+
+	"github.com/osbuild/image-builder/internal/db"
+)
+
+const (
+	day                  time.Duration = 24 * time.Hour
+	week                 time.Duration = 7 * day
+	defaultSlidingWindow time.Duration = 2 * week
+	defaultQuota         int           = 100
+)
+
+type quota struct {
+	Quota         int           `json:"quota"`
+	SlidingWindow time.Duration `json:"slidingWindow"`
+}
+
+// Returns true if the number of requests made by accountNumber during a sliding window is inferior to a threshold.
+// The duration of the sliding window and the value of the threshold can be set in /config/accounts_quotas.json
```

Comment:
I think we need to make the exact path configurable via the env, so `QUOTA_FILE`  (in internal/config) or something. And everything tends to be in `/app/` (see the dockerfile).

Reply:
Good idea!